docs/oauth2.md

```diff
@@ -538,13 +664,94 @@ In the following RAR example
+
+```ini
+auth_oauth2.issuer = http//some_idp_url
+auth_oauth2.scope_prefix = rabbitmq.
+
+auth_oauth2.resource_servers.1.id = rabbit_prod
+auth_oauth2.resource_servers.2.id = rabbit_dev
+auth_oauth2.resource_servers.2.scope_prefix = dev-rabbitmq.
+```
+
+All resource servers share the settings you have set so far under `auth_oauth2.` such as `scope_prefix`. However, they can override them. Here are the list of settings a resource server can override:
+- `id` - This is the actual resource identifier carried in the `audience` field of a token. If omitted, the value is the index, e.g. given `auth_oauth2.resource_servers.prod.scope_prefix` setting, the `id` would be `prod`.
+- `scope_prefix`
+- `additional_scopes_key`
+- `resource_server_type`
+- `oauth_provider_id` - This is the identifier of the OAuth provider, configured in RabbitMQ, which provides all the settings to contact the authorization server in order to discover all its endpoints, such as the `jwks_uri` to download the signing keys to validate the token. If this setting is omitted, RabbitMQ looks up the default Authorization Provider's id in the setting `auth_oauth2.default_oauth_provider`. And if it is also omitted, RabbitMQ uses `auth_oauth2.issuer` or `auth_oauth2.jwks_url` to download the signings keys to validate the token.
+
+The list of supported resource servers is the combination of `auth_oauth2.resource_servers` and `auth_oauth2.resource_server_id`. You can use both or only one of them.
+
+**NOTE**: There is an [example](./oauth2-examples-multiresource) that demonstrate multiple OAuth 2 resources.
+
+### Multiple OAuth Providers {#multiple-oauth-providers}
+
+It only makes sense to set multiple OAuth Providers if there are [multiple resources configured](#multiple-resource-servers).
+
+This is the configuration used in the previous section but modified to use multiple OAuth providers:
+
+```ini
+auth_oauth2.scope_prefix = rabbitmq.
+
+auth_oauth2.resource_servers.1.id = rabbit_prod
+auth_oauth2.resource_servers.1.oauth_provider_id = prod
+auth_oauth2.resource_servers.2.id = rabbit_dev
+auth_oauth2.resource_servers.2.oauth_provider_id = dev
+auth_oauth2.resource_servers.2.scope_prefix = dev-rabbitmq.
+
+auth_oauth2.oauth_providers.prod.issuer = https://rabbit_prod:8080
+auth_oauth2.oauth_providers.prod.https.cacertfile = /opts/certs/prod.pem
+auth_oauth2.oauth_providers.dev.issuer = https://rabbit_dev:8080
+auth_oauth2.oauth_providers.dev.https.cacertfile = /opts/certs/dev.pem
+```
+
 ## Examples {#examples}
 
 The [RabbitMQ OAuth 2.0 Auth Backend Examples](./oauth2-examples) contains many example configuration files which can be used to set up several OAuth 2.0 providers, including UAA, Auth0, and Azure, and issue tokens, which can be used to access RabbitMQ resources.
```

Comment:
does "Backend Examples" need to be in CAPS, I would change to [RabbitMQ OAuth 2.0 Auth backend examples](./oauth2-examples)

Change "contains" to "contain"
add a comma after "files" and before "which"...I would say "that" instead of "which"